pilot/pkg/bootstrap/server.go

```diff
@@ -1334,3 +1341,27 @@ func (s *Server) serveHTTP() error {
 	s.httpAddr = httpListener.Addr().String()
 	return nil
 }
+
+func serviceUpdateNeedsPush(prev, curr *model.Service) bool {
+	if prev == nil {
+		return true
+	}
+	// If current/previous metadata has "*istio.io" label/annotation, just push
+	for label := range prev.Attributes.Labels {
+		if strings.Contains(label, "istio.io") {
+			return true
+		}
+	}
+	for label := range curr.Attributes.Labels {
+		if strings.Contains(label, "istio.io") {
+			return true
+		}
+	}
+	// Exclude labels and compare so that we do not push on unwanted label changes.
+	currclone := curr.DeepCopy()
+	prevclone := prev.DeepCopy()
+	currclone.Attributes.Labels = map[string]string{}
+	prevclone.Attributes.Labels = map[string]string{}
+	// TODO: Make this more efficient by comparing field by field.
+	return !reflect.DeepEqual(prevclone, currclone)
```

Comment:
This is costy

Reply:
I know. I will replace with a more optimized version in a follow-up if we are good with this change.